src/main/asciidoc/en/Ch05_DBAccess.adoc

```diff
@@ -317,141 +334,726 @@ that uses the mechanism described
+    // (8)
+    @Inject
+    ItemStreamReader<SalesPlanSummary> reader;
+    
+    @Override
+    public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
+        SalesPlanSummary item = null;
+
+        List<SalesPlanSummary> items = new ArrayList<>(CHUNK_SIZE);
+        int errorCount = 0;
+
+        try {
+            // (9)
+            reader.open(chunkContext.getStepContext().getStepExecution().getExecutionContext());
+            while ((item = reader.read()) != null) { // (10)
+                if (item.getAmount().signum() == -1) {
+                    logger.warn("amount is negative. skip item [item: {}]", item);
+                    errorCount++;
+                    continue;
+                }
+
+                // omitted.
+            }
+            
+            // catch block is omitted.
+        } finally {
+            // (11)
+            reader.close();
+        }
+    }
+    // omitted.
+    
+    return RepeatStatus.FINISHED;
+}
+----
+
+[cols="10,90", options="header"]
+.Description
+|===
+|Sr. No.
+|Description
+
+|(1)
+|Register Mapper XML.
+
+|(2)
+|Define ``MyBatisCursorItemReader``.
+
+|(3)
+|Specify the SQL ID defined in (6) with ``namespace`` + ``<method name>`` of (5) to the property of ``queryId``.
+
+|(4)
+|Specify ``SqlSessionFactory`` of the database to be accessed in ``sqlSessionTemplate-ref`` property.
+
+|(5)
+|Define Mapper XML. Match the value of namespace with the FQCN of the interface.
+
+|(6)
+|Define SQL.
+
+|(7)
+|Define the method corresponding to the SQL ID defined in (6) for the interface.
+
+|(8)
+|Assign ``@Inject`` annotation and inject ``ItemStreamReader`` implementation. +
+Since there is a need to open/close the target resource,
+inject implementation in ``ItemStreamReader`` interface wherein open/close method of resource is added to ``ItemReader``.
+
+|(9)
+|Open input resource.
+
+|(10)
+|Read input data one by one.
+
+|(11)
+|Close input resource. +
+Be sure to close the resource. When an exception occurs, the transaction of entire tasklet is rolled back,
+stack trace  of exception is output and the job ends abnormally. Therefore, exception handling should be implemented whenever required.
+
+|===
+
+[[Ch05_DBAccess_HowToUse_Input_MapperInterface]]
+==== Mapper interface (Input)
+
+Use Mapper interface for referring to the database except for ItemReader. +
+Here, database reference using Mapper interface is described.
+
+[[Ch05_DBAccess_HowToUse_Input_MapperInterface_Overview]]
+Functional overview::
+
+In using the Mapper interface, {batch5_shortname} has the following restrictions.
+
+.The available points of Mapper interface.
+
+It is possible to use ``MyBatisBatchItemWriter`` by Inject, but in that case Mapper interface itself can be processed with ``BATCH`` setting.
+  In other words, there is basically no need to use ``MyBatisBatchItemWriter`` by Inject.
+
+[[Ch05_DBAccess_HowToUse_Output_MapperInterface_Chunk]]
```

Comment:
@bttendulkara L-650 to L-653
Do you really require this statements here. Please confirm